### PR TITLE
Add cron & dispatch to CI + bump python range

### PR DESCRIPTION
**Pull request by IAlibay**

### CODE REVIEW

**Comment by IAlibay** [comment]
Failure is unrelated to this PR (see #102 )

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -24,19 +28,19 @@ jobs:
...
-            python-version: "3.11"
+            python-version: "3.12"
 
     steps:
       - uses: actions/checkout@v4
       - name: Get current date
         id: date
         run: echo "date=$(date +%Y-%m-%d)" >> "${GITHUB_OUTPUT}"
-      - uses: mamba-org/setup-micromamba@v1
+      - uses: mamba-org/setup-micromamba@v2
```
Comment (atravitz) [inline .github/workflows/ci.yaml:43]: we're still using micromamba v1 for openfe - do we want to bump that to v2? (personally /coI would prefer to)

**Review by atravitz — APPROVED** [review]
fine to merge, but I have a couple clarifying questions.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (IAlibay) [reply .github/workflows/ci.yaml:43]: We should - it's not going to break today, but it's always nice to do it.

**Pull request merged**